lib/resty/upload.lua

```diff
@@ -62,20 +65,35 @@ function _M.new(self, chunk_size, max_line_size)
         return nil, err
     end
 
+    if restore_body_buffer then
+        ngx_init_body(chunk_size)
+    end
+
     local read2boundary, err = sock:receiveuntil("--" .. boundary)
     if not read2boundary then
+
+        if restore_body_buffer then
+            ngx_finish_body()
+        end
+
         return nil, err
     end
 
     local read_line, err = sock:receiveuntil("\r\n")
     if not read_line then
+
+        if restore_body_buffer then
+            ngx_finish_body()
```

Comment:
The previous read, on line 72, was successful, but we didn't write the request body back, is this intended? Maybe there are many similar cases in code, but no easy way to solve? E.g. on errors we don't write the request body back.

Reply:
The sock:receiveuntil returns an iterator Lua function and doesn't actually read data from the buffer so there is noting to restore.